### PR TITLE
Add DeletedAware interface and support

### PR DESCRIPTION
**Pull request by lchanmann**
Address: https://github.com/nanoPayinc/NANOPAY/issues/5600

### CODE REVIEW

File: src/foam/nanos/auth/DeletedAwareDAO.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  name: 'DeletedAwareDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'DAO that sets deleted property',
+
+  methods: [
+    {
+      name: 'remove_',
+      javaCode: `
+        if ( obj instanceof DeletedAware ) {
```
Comment (jlhughes) [inline src/foam/nanos/auth/DeletedAwareDAO.js:18]: Suspect you'll need to clone here.

**Review by jlhughes — CHANGES_REQUESTED** [review]

File: src/foam/nanos/auth/DeletedAwareDAO.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/nanos/auth/DeletedAwareDAO.js:18]: Fixed.

**Comment by lchanmann** [comment]
@jlhughes please review again.

**Comment by jlhughes** [comment]
Can you create a test case for this please. Thanks.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**